docs/references/ic-interface-spec.md

```diff
@@ -1757,9 +1757,9 @@ There must be at most one call to `ic0.call_on_cleanup` between `ic0.call_new` a
 
     This deducts `MAX_CYCLES_PER_RESPONSE` cycles from the canister balance and sets them aside for response processing.
 
-    If the function returns `0` as the `err_code`, the IC was able to enqueue the call. In this case, the call will either be delivered, returned because the destination canister does not exist, returned due to a lack of resources within the IC, or returned because of an out of cycles condition. This also means that exactly one of the reply or reject callbacks will be executed.
+    The returned `err_code` is always one of `0`, `1`, `2`, and `3`. An `err_code` of `0` means that no error occurred and that the IC could enqueue the call. In this case, the call will either be delivered, returned because the destination canister does not exist, returned due to a lack of resources within the IC, or returned because of an out of cycles condition. This also means that exactly one of the reply or reject callbacks will be executed.
```

Comment:
>  The returned `err_code` is always one of `0`, `1`, `2`, and `3`.

Are we sure that the implementation does not start producing the new code `6` for "unknown" in some cases in the future?

Reply:
Discussion in the interface spec meeting:
- Currently there is inconsistent behavior in messages sent to the IC management canister, where in case of an invalid candid payload, the canister would trap. (Which is inconsistent with what happens if invalid candid is sent to another canister.)
- For the same case for ingress messages, we would fail with a `4`.
- If we ever wanted to clean up this behavior, this would mean we would want to return a `4` synchronously.

Reply:
> Currently there is inconsistent behavior in messages sent to the IC management canister, where in case of an invalid candid payload, the canister would trap. (Which is inconsistent with what happens if invalid candid is sent to another canister.)

This hypothesis of mine doesn't hold: if an invalid candid payload is sent to the IC management canister, then an asynchronous reject with reject code 4 is produced.

Nevertheless, if we include the reject code 3 as a possible reject code for synchronous routing errors in the future, we might also need to include the reject code 4 for synchronous routing errors to the IC management canister due to invalid candid payload.